examples/data_interpreter_mulit-agent/di_agents.py

```diff
@@ -0,0 +1,414 @@
+    PlannerAgent is responsible for decomposing complex tasks into manageable
+    subtasks.
+
+    This agent takes an overall task and breaks it down into subtasks that can
+    be solved using available tools or code execution. It ensures that each
+    subtask is appropriately sized and prioritizes using tools over code
+    execution when possible.
+    """
+
+    def __init__(
+        self,
+        name: str,
+        sys_prompt: str,
+        model_config_name: str,
+        service_toolkit: ServiceToolkit,
+    ):
+        super().__init__(
+            name=name,
+            sys_prompt=sys_prompt,
+            model_config_name=model_config_name,
+        )
+        self.service_toolkit = service_toolkit
+
+    def __call__(self, x: Msg) -> List[Dict[str, Any]]:
+        return self.plan(x)
+
+    def plan(self, x: Msg) -> List[Dict[str, Any]]:
+        """
+        Decompose the task provided in the message into subtasks.
+
+        Args:
+            x (Msg): Message containing the task to be decomposed.
+
+        Returns:
+            List[Dict[str, Any]]: List of subtasks as dictionaries.
+        """
+        messages = x.content
+        subtasks = self._decompose_task(messages)
+        return subtasks
+
+    def _decompose_task(
+        self,
+        task: str,
+        max_tasks: int = 5,
+    ) -> List[Dict[str, Any]]:
+        # Implement task decomposition
+        message = [
+            {
+                "role": "user",
+                "content": f"""
+                    Task: {task}
+                    - Given the task above, break it down into subtasks with dependencies if it is sufficently complex to be solved in one go.
+                    - Every subtask should be solvable through either executing code or using tools. The information of all the tools available are here:
+                    {self.service_toolkit.tools_instruction}
+                    - The subtask should not be too simple. If a task can be solve with a single block of code in one go, it should not be broken down further. Example: a subtask cannot be simply installing or importing libraries.
+                    - Prioritze using other tools over `execute_python_code` and take the tools available into consideration when decomposing the task. Provide a JSON structure with the following format for the decomposition:
+                        ```json
+                        [
+                            {{
+                                "task_id": str = "unique identifier for a task in plan, can be an ordinal",
+                                "dependent_task_ids": list[str] = "ids of tasks prerequisite to this task",
+                                "instruction": "what you should do in this task, one short phrase or sentence",
+                                "task_type": "type of this task, should be one of Available Task Types",
+                                "task_type": "type of this task, should be one of Available Task Types",
```

Comment:
1) Duplicate lines?
2) what are the `Available Task Types`?

Reply:
1. Duplicates removed.
2. Since this framework is designed to be generic, there are actually no pre-defined Available Task Types. `task_type` is optional since its only purpose at the moment is to give the solver a high-level idea of what kind of tool to employ. I have corrected the relevant prompts.